src/ngraph/runtime/gpu/cudnn_emitter.cpp

```diff
@@ -166,3 +170,148 @@ std::vector<int> cudnn_util::compute_strides(const std::vector<int>& dim)
 
     return strides;
 }
+
+cudnnTensorDescriptor_t cudnn_util::tensor_descriptor_4d_from_shape(const Shape& shape)
+{
+    cudnnTensorDescriptor_t desc;
+    cudnnCreateTensorDescriptor(&desc);
+    cudnnSetTensor4dDescriptor(
+        desc, CUDNN_TENSOR_NCHW, CUDNN_DATA_FLOAT, shape[0], shape[1], shape[2], shape[3]);
+    return desc;
+}
+
+cudnnTensorDescriptor_t cudnn_util::tensor_descriptor_Nd_from_shape(const Shape& shape)
+{
+    std::vector<int> dimensions(shape.size());
+    for (auto i = 0u; i < shape.size(); i++)
+    {
+        dimensions[i] = static_cast<int>(shape[i]);
+    }
+    cudnnTensorDescriptor_t desc;
+    cudnnCreateTensorDescriptor(&desc);
+    cudnnSetTensorNdDescriptor(desc,
+                               CUDNN_DATA_FLOAT,
+                               dimensions.size(),
+                               dimensions.data(),
+                               cudnn_util::compute_strides(dimensions).data());
+    return desc;
+}
+
+size_t CUDNNEmitter::build_pooling(const GPURuntimeContext* ctx,
+                                   const cudnnPoolingMode_t& pool_op,
+                                   const Prop& direction,
+                                   const ngraph::Shape& input_shape,
+                                   const ngraph::Shape& output_shape,
+                                   const ngraph::Strides& window_strides,
+                                   const ngraph::Shape& window_shape,
+                                   const ngraph::Shape& padding_below,
+                                   const ngraph::Shape& padding_above)
+{
+    // construct hash to determine if kernel needs to be emitted
+    // or if it already exists in the primitive list
+    std::stringstream ss;
+    ss << "pool_op" << pool_op << "_dir" << static_cast<int>(direction) << "_i" << join(input_shape)
```

Comment:
My thought is if the parameter is not hard coded into kernel, it does not need to go into hash, since they can share same primitive.